examples/cluster/main.tf

```diff
@@ -1,74 +1,53 @@
-##tag with kubernetes.io/cluster/CLUSTER_NAME
+data "aws_caller_identity" "current" {}
 
-data "aws_vpc" "default" {
-  default = true
+data "aws_availability_zones" "available" {
+  state = "available"
 }
 
-locals {
-  cluster_name = "test-cluster-${uuid()}"
-  role_name    = "eks-test-role"
-  cidr_block1  = cidrsubnet(data.aws_vpc.default.cidr_block, 8, 65)
-  cidr_block2  = cidrsubnet(data.aws_vpc.default.cidr_block, 8, 70)
-  cidr_blocks  = [local.cidr_block1, local.cidr_block2]
-}
-
-module "eks_role" {
-  source                = "boldlink/iam-role/aws"
-  version               = "1.0.0"
-  name                  = local.role_name
-  assume_role_policy    = <<POLICY
-{
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Effect": "Allow",
-      "Principal": {
-        "Service": "eks.amazonaws.com"
-      },
-      "Action": "sts:AssumeRole"
-    }
-  ]
-}
-POLICY
-  description           = "Role for eks cluster permissions"
-  force_detach_policies = true
-}
+data "aws_region" "current" {}
 
-module "eks_cluster_subnets" {
-  source             = "boldlink/subnet/aws"
-  version            = "1.0.0"
-  vpc_id             = data.aws_vpc.default.id
-  availability_zones = ["eu-west-1a", "eu-west-1b"]
-  cidr_blocks        = local.cidr_blocks
-  tags = {
-    Name = "kubernetes.io/cluster/${local.cluster_name}"
-  }
+locals {
+  name               = "boldlink-test"
+  cidr_block         = "10.0.0.0/16"
+  tag_env            = "dev"
+  cluster_name       = "dynamic-eks-cluster"
+  eks_public_subnet1 = cidrsubnet(local.cidr_block, 8, 50)
+  eks_public_subnet2 = cidrsubnet(local.cidr_block, 8, 55)
+  eks_public_subnet3 = cidrsubnet(local.cidr_block, 8, 60)
+  eks_public_subnets = [local.eks_public_subnet1, local.eks_public_subnet2, local.eks_public_subnet3]
+
+  az1 = data.aws_availability_zones.available.names[0]
+  az2 = data.aws_availability_zones.available.names[1]
+  az3 = data.aws_availability_zones.available.names[2]
+  azs = [local.az1, local.az2, local.az3]
+}
+
+module "eks_network" {
```

Comment:
Just for the example, will be good to have the property `cluster_name` to this VPC. Otherwise it will come with "kubernetes.io/cluster/none" as the tag, making this useless when deploying load balancer with the cluster name